disperser/dataapi/nonsigner_handler.go

```diff
@@ -0,0 +1,236 @@
+package dataapi
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigenda/core/eth"
+)
+
+func (s *server) getOperatorNonsigningRate(ctx context.Context, intervalSeconds int64) (*OperatorsNonsigningPercentage, error) {
+	batches, err := s.subgraphClient.QueryBatchNonSigningInfoInInterval(ctx, intervalSeconds)
+	if err != nil {
+		return nil, err
+	}
+	if len(batches) == 0 {
+		return &OperatorsNonsigningPercentage{}, nil
+	}
+
+	// Get the block interval of interest [startBlock, endBlock].
+	startBlock := batches[0].ReferenceBlockNumber
+	endBlock := batches[0].ReferenceBlockNumber
+	for i := range batches {
+		if startBlock > batches[i].ReferenceBlockNumber {
+			startBlock = batches[i].ReferenceBlockNumber
+		}
+		if endBlock < batches[i].ReferenceBlockNumber {
+			endBlock = batches[i].ReferenceBlockNumber
+		}
+	}
+
+	// Get the nonsigner (in operatorId) list.
+	nonsigners, err := getNonSigners(batches)
+	if err != nil {
+		return nil, err
+	}
+	if len(nonsigners) == 0 {
+		return &OperatorsNonsigningPercentage{}, nil
+	}
+
+	// Get the address for the nonsigners (from their operatorIDs).
+	// nonsignerAddresses[i] is the address for nonsigners[i].
+	nonsignerAddresses, err := s.transactor.BatchOperatorIDToAddress(ctx, nonsigners)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a mapping from address to operatorID.
+	nonsignerAddressToId := make(map[string]core.OperatorID)
+	for i := range nonsigners {
+		nonsignerAddressToId[nonsignerAddresses[i].Hex()] = nonsigners[i]
+	}
+
+	// Create operators' quorum intervals.
+	operatorQuorumIntervals, err := s.createOperatorQuorumIntervals(ctx, nonsigners, nonsignerAddressToId, startBlock, endBlock)
+	if err != nil {
+		return nil, err
+	}
+
+	// Compute num batches failed, where numFailed[op][q] is the number of batches
+	// failed to sign for operator "op" and quorum "q".
+	numFailed := computeNumFailed(batches, operatorQuorumIntervals)
+
+	// Compute num batches responsible, where numResponsible[op][q] is the number of batches
+	// that operator "op" and quorum "q" are responsible for.
+	numResponsible := computeNumResponsible(batches, operatorQuorumIntervals)
+
+	// Compute the nonsigning rate for each <operator, quorum> pair.
+	operators := make([]*OperatorNonsigningPercentageMetrics, 0)
+	for op, val := range numResponsible {
```

Comment:
nit: val doesn't mean anything. Can we name it something like numResponsibleByQuorum?

Reply:
done